File: art-gallery/src/main/java/com/launchcode/artgallery/models/Artwork.java
package com.launchcode.artgallery.models;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@Entity
public class Artwork extends AbstractEntity {

    @NotBlank(message = "Title is required.")
    private String title;

    @OneToOne(cascade = CascadeType.ALL)
    @Valid
    private ArtworkDetails details;

    @ManyToOne
    @NotNull(message = "Artist is required.")
    private Artist artist;

    private Style style;

    public Artwork() {}

    public Artwork(String title, ArtworkDetails details, Artist artist, Style style) {
        this.title = title;
        this.details = details;
        this.artist = artist;
        this.style = style;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public ArtworkDetails getDetails() {
        return details;
    }

    public void setDetails(ArtworkDetails details) {
        this.details = details;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }

    public Style getStyle() {
        return style;
    }

    public void setStyle(Style style) {
        this.style = style;
    }

    @Override
    public String toString() {
        return title + " (" + artist + ", " + details.getYearCreated() + ")";
    }

}

File: front-end/react/art-gallery-public/src/classes/Artwork.js
export default class Artwork {

  constructor(id, title, artist, styles, details) {
    this.id = id;
    this.title = title;
    this.artist = artist; // Artist object
    this.styles = styles; // array of Style objects
    this.details = details; // ArtworkDetails object
  }

  getFormattedStyles() {
    let styleNames = "";
    for (let i=0; i < this.styles.length; i++) {
        styleNames += this.styles[i].name;
        if (i < this.styles.length - 1) {
            styleNames += ", ";
        }
    }
    return styleNames;
  }

}